src/System.Windows.Forms/src/System/Windows/Forms/TaskDialogIcon.cs

```diff
@@ -93,47 +93,83 @@ public class TaskDialogIcon
 #pragma warning restore IDE1006 // Naming Styles
 
         private readonly TaskDialogStandardIcon? _standardIcon;
-
+        private readonly Icon? _ownedIcon;
         private readonly IntPtr? _iconHandle;
 
+        /// <summary>
+        ///   Initializes a new instance of the <see cref="TaskDialogIcon"/> class from an
+        ///   <see cref="Bitmap"/> instance.
+        /// </summary>
+        /// <param name="image">The <see cref="Bitmap"/> instance.</param>
+        /// <exception cref="ArgumentNullException"><paramref name="image"/> is <see langword="null"/>.</exception>
+        /// <remarks>
+        ///   <para>
+        ///     The <see cref="Icon"/> instance from which this <see cref="TaskDialogIcon"/>
+        ///     instance is created must not be disposed while the icon is shown in the task
+        ///     dialog.
+        ///   </para>
+        /// </remarks>
+        public TaskDialogIcon(Bitmap? image)
+            : this(BitmapToIcon(image ?? throw new ArgumentNullException(nameof(image))), true)
```

Comment:
If the method throws an `ArgumentNullException` when `null` is supplied, I think the parameter should not be marked as nullable (`Bitmap` rather than `Bitmap?`).
```suggestion
        public TaskDialogIcon(Bitmap image)
            : this(BitmapToIcon(image ?? throw new ArgumentNullException(nameof(image))), true)
```